Source/WebCore/platform/graphics/MediaSourcePrivate.h

```diff
@@ -136,12 +140,16 @@ class WEBCORE_EXPORT MediaSourcePrivate
 
 private:
     mutable Lock m_lock;
+    void updateTracksType();
+
     MediaTime m_duration WTF_GUARDED_BY_LOCK(m_lock) { MediaTime::invalidTime() };
     PlatformTimeRanges m_buffered WTF_GUARDED_BY_LOCK(m_lock);
     PlatformTimeRanges m_liveSeekable WTF_GUARDED_BY_LOCK(m_lock);
     std::atomic<bool> m_streaming { false };
     std::atomic<bool> m_streamingAllowed { false };
     MediaTime m_timeFudgeFactor;
+    HashMap<SourceBufferPrivate*, TracksType> m_tracksTypes WTF_GUARDED_BY_CAPABILITY(m_dispatcher.get());
+    std::atomic<TracksType> m_tracksType;
```

Comment:
It does not read great to have m_tracksTypes and m_tracksType.
Maybe m_tracksCombinedType?